### PR TITLE
Add coolant control

### PR DESCRIPTION
**Pull request by benagricola**
Add the settings required for coolant control.

We add 3 new variables for storing the pin assigned to Air, Mist and Flood.

### CODE REVIEW

**Comment by benagricola** [comment]
@bdistin I've reworked your implementation slightly so `M9` now handles controlling all coolant outputs at once. By default it will turn all coolant outputs off, but when run with `R1` it will restore the previous values.

`pause.g` stores all the values now as this might be useful for other settings in future, but we only restore the ones configured as coolant, just in case the operator has something else set up on another `gpOut` pin.

File: sys/pause.g [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ; pause.g - PAUSE CURRENT JOB
 
-; Raise the spindle to the top of
-; the Z axis and stop the spindle
-; but do not move the table.
+; Save pre-pause state of all general purpose
+; output pins.
+while { iterations < #state.gpOut }
+    set global.mosPS[iterations] = state.gpOut[iterations].pwm
+
```
Comment (bdistin) [inline sys/pause.g:7]: The states have been saved, but coolant options haven't been stopped.

**Review by bdistin — CHANGES_REQUESTED** [review]
That's all I see from a manual read-through.

File: sys/pause.g (hunk 1 above)
Comment (benagricola) [reply sys/pause.g:7]: Thanks. I'm pretty sure this is handled in `G27` which is called straight after the pin states are saved, am I missing something? (there's a call to `M9` before the tool lifts away from the workpiece).

**Pull request merged**